src/Compilers/Core/Portable/InternalUtilities/ISetExtensions.cs

```diff
@@ -17,6 +18,17 @@ public static bool AddAll<T>(this ISet<T> set, IEnumerable<T> values)
             return result;
         }
 
+        public static bool AddAll<T>(this ISet<T> set, ImmutableArray<T> values)
+        {
+            var result = false;
+            foreach (var v in values)
+            {
+                result |= set.Add(v);
```

Comment:
I am not trying to redefine the result, just adding a more efficient overload for an ImmutableArray. Either behavior would be fine, depending on consumer's needs. For the purposes of this PR, I do not care what is the result.